### PR TITLE
renamed CourseSelection -> Timeline, TermSelection -> TermRecord

### LINKED ISSUE #20: Rename CourseSelection/TermSelection to Timeline/TermRecord
Will also (depending on the name) rename the entire app to this name.

### PR DESCRIPTION
**Pull request by infinitecold**
(will close #20) also moved TermRecord into terms package

### CODE REVIEW

**Comment by gzcharleszhang** [comment]
how ru so fast

File: internal/app/components/courses/courseRecords_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"time"
 )
 
-func TestCourseRecords_ToCourseIdMap(t *testing.T) {
+func TestCourseRecord_ToCourseIdMap(t *testing.T) {
```
Comment (gzcharleszhang) [inline internal/app/components/courses/courseRecords_test.go]: why are we changing the name of this func?

**Comment by infinitecold** [comment]
😏

File: internal/app/components/courses/courseRecords_test.go (hunk 1 above)
Comment (infinitecold) [reply internal/app/components/courses/courseRecords_test.go]: consistency lmao: `Test(struct name)_(func name)`

Comment (infinitecold) [reply internal/app/components/courses/courseRecords_test.go]: oh shoot i goofed

File: internal/app/components/timelines/timelines.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	Name        TimelineName
+	TermRecords []*terms.TermRecord
+	Plans       plans.Plans
+}
+
+func newTimelineId() TimelineId {
+	return TimelineId(xid.New().String())
+}
+
+func NewTimeline(name TimelineName) *Timeline {
```
Comment (gzcharleszhang) [inline internal/app/components/timelines/timelines.go]: can you add a todo here for using course history when adding a new timeline? this method could prob just take `CourseRecords` as an additional parameter

Comment (gzcharleszhang) [reply internal/app/components/timelines/timelines.go]: just a TODO is fine for this PR tho

File: internal/app/components/courses/courseRecords_test.go (hunk 1 above)
Comment (gzcharleszhang) [reply internal/app/components/courses/courseRecords_test.go]: LOL all good

File: internal/app/components/terms/terms_test.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ import (
 	"testing"
 )
 
-func TestNewTerm(t *testing.T) {
+func Test_NewTerm(t *testing.T) {
```
Comment (gzcharleszhang) [inline internal/app/components/terms/terms_test.go]: I think `TestNewTerm` is the right name, if it's just a function it's just `Test` + `FuncName`. If it's a method then it's `Test` + `StructName` + `_MethodName`

Comment (infinitecold) [reply internal/app/components/terms/terms_test.go]: why am i incompetent

Comment (gzcharleszhang) [reply internal/app/components/terms/terms_test.go]: that's why you use gotests to generate test funcs! they never get the name wrong

Comment (infinitecold) [reply internal/app/components/terms/terms_test.go]: ok then explain Test_isPrereqSatisfied

Comment (gzcharleszhang) [reply internal/app/components/terms/terms_test.go]: ok idk wtf is happening

Comment (gzcharleszhang) [reply internal/app/components/terms/terms_test.go]: maybe it's because it's a private func so it wants to maintain the lowercase

Comment (gzcharleszhang) [reply internal/app/components/terms/terms_test.go]: yeah can confirm that is the case, the more you know! LOL

Comment (infinitecold) [reply internal/app/components/terms/terms_test.go]: go is dumb

Comment (gzcharleszhang) [reply internal/app/components/terms/terms_test.go]: at least we have consistency in naming convention

**Pull request merged**